Sources/IO/Core/HttpDataSetReader/index.d.ts

```diff
@@ -148,6 +148,33 @@ export interface vtkHttpDataSetReader extends vtkHttpDataSetReaderBase {
 	 */
 	getUrl(): string;
 
+	/**
+	 * Gets if arrays are stored in cache.
+	 */
+	getCacheArrays(): boolean;
+
+	/**
+	 * Sets if arrays are stpred in cache.
+	 */
+	setCacheArrays(value: boolean): void;
+
+
+	/**
+	 * Gets the maximum size cached arrays are allowed to occupy.
+	 * Size is given in MiB.
+	 * If cache size is exceeded, the arrays that where not accessed
+	 * the longest are removed.
+	 */
+	getMaxCacheSize(): number;
+
+	/**
+	 * Sets the maximum size cached arrays are allowed to occupy.
+	 * Size is given in MiB.
+	 * If cache size is exceeded, the arrays that where not accessed
+	 * the longest are removed.
+	 */
+	setMaxCacheSize(value: numnber): void;
```

Comment:
Maybe we could have:
 `maxCacheSize = undefined` => unlimited cache size
 `maxCacheSize = none` => no cache usage

That way you do not need to create a separate `cacheArrays` variable with accessors.

Reply:
I see your point but there is one question about ```maxCacheSize = none```. As far as I know ```none``` is not a built-in value in Javascript therefore, how should I add it? One option would be to allow strings to be set:
```javascript
setMaxCacheSize('none')  // disable caching
```
However, this means that we allow different datatypes for one property (nullable/undefinable numbers and string). I know that this is possible and probably fine in Javascript but as a developer more used to languages like C# or C++ this is a bit icky. How about we rename the ```cacheArrays``` property to ```arrayCachingEnabled```?

Reply:
I have implemented my idea and I think using ```arrayCachingEnabled``` makes it more obvious what the intention is than the old name or using ```setMaxCacheSize('none'). Interested what your opinion is @finetjul.

Reply:
By `none` I meant `null`

Reply:
Makes sense :)
But even with that in mind, I would still argue to use a separate property to control this. I would say the behavior
```
null -> caching disabled
undefined -> unlimited cache size
```
Is not that intuitive for people unfamiliar with the implementation.

Reply:
Fair enough. I'm fine with `setArrayCachingEnabled`.
But what behavior will you have if maxCacheSize is undefined/null ? caching will be disabled ?

Reply:
The property has been removed and the following has been implemented:
```
-1 -> unlimited cache
0 -> cache disabled
null -> cache disabled
undefined -> cache disabled
```